macros/snowplow_media_player_base_events_this_run/get_context_fields.sql

```diff
@@ -0,0 +1,29 @@
+{#
+Copyright (c) 2022-present Snowplow Analytics Ltd. All rights reserved.
+This program is licensed to you under the Snowplow Personal and Academic License Version 1.0,
+and you may not use this file except in compliance with the Snowplow Personal and Academic License Version 1.0.
+You may obtain a copy of the Snowplow Personal and Academic License Version 1.0 at https://docs.snowplow.io/personal-and-academic-license-1.0/
+#}
+
+{% macro get_context_fields(fields, enabled, context, prefix=None) %}
+  {{ return(adapter.dispatch('get_context_fields', 'snowplow_media_player')(fields, enabled, context, prefix)) }}
+{% endmacro %}
+
+{% macro default__get_context_fields(fields, enabled, context, prefix) %}
+
+  {%- for f in fields %}
+    {% if '_' in f.get('field') %}
+      {{ exceptions.raise_compiler_error(f ~ ' field value should be in Camel case and not contain any "_" characters') }}
+    {% endif %}
+  {% endfor %}
+
```

Comment:
Couldn't the schema itself contain snake case fields and they would be snake case in snowflake then?

Reply:
Yeah true, I'll remove this check. For Snowflake we take the field name as is, for other warehouses we snakeify the field name which I think matches how the loader behaves.